### PR TITLE
Update Date docs. Add Date.parse and Date.p.getUTCDay

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: src/stdlib/date.md [hunk 1]
```diff
@@ -34,9 +34,14 @@ The Date API is still preliminary with only a subset of properties and methods i
   ): i64
   ```
   Returns the UTC timestamp in milliseconds of the specified date.
-  
+
+* ```ts
+  function parse(dateString: string): Date
+  ```
+  Parse a Date object from a string (ISO 8601 format only).
```
Comment (saulecabrera) [inline src/stdlib/date.md]:
```suggestion
  Parses a Date object from a string (ISO 8601 format only).
```

File: src/stdlib/date.md [hunk 2]
```diff
@@ -81,7 +86,12 @@ The Date API is still preliminary with only a subset of properties and methods i
   function setUTCDate(value: i32): i32
   ```
   Sets the day of the month according to universal time.
-  
+
+* ```ts
+  function getUTCDay(): i32
+  ```
+  Gets the day of the week in the specified date according to universal time, where 0 represents Sunday.
```
Comment (saulecabrera) [inline src/stdlib/date.md:93]:
```suggestion
  Returns the day of the week in the specified date according to universal time, where 0 represents Sunday.
```

Comment (MaxGraey) [reply src/stdlib/date.md:93]: I just try to follow already existing conventions which use `Gets` and `Sets` as starting word for rest of set/set methods.

Comment (saulecabrera) [reply src/stdlib/date.md:93]: No strong opinions, looks good to me either way 👍  I was just following https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCDay

Comment (MaxGraey) [reply src/stdlib/date.md:93]: yeah, I also use MDN as reference.

**Pull request merged**